pkg/genericapiserver/server/openapi/openapi.go

```diff
@@ -79,6 +81,15 @@ func (o *openAPI) init(webServices []*restful.WebService) error {
 			return r.Operation, nil, nil
 		}
 	}
+	if o.config.GetDefinitionName == nil {
+		o.config.GetDefinitionName = func(_, name string) (string, []string) {
+			return name[strings.LastIndex(name, "/")+1:], nil
+		}
+	}
+	o.definitions = o.config.GetDefinitions(func(name string) spec.Ref {
+		defName, _ := o.config.GetDefinitionName(o.servePath, name)
+		return spec.MustCreateRef("#/definitions/" + openapi.EscapeJsonPointer(defName))
```

Comment:
"." Would be the most used separator in the widest range of languages.  If we do that I think we should also do domain reversing (github.com -> com.github).  Go's type naming systems is fairly obscure.